### ISSUE TITLE
Consider making "output adapters" explicit post-processing steps

### ISSUE THREAD

**Issue by maxpumperla** [KonduitAI/konduit-serving#113]
Our concept of `OutputAdapter` is an important part of our pipelines, as it changes the output given certain `predictionType`s. However, the way this has to be configured takes getting used to and is certainly not very user-friendly (e.g. what do I have to set my output format and prediction type to to get the result I want requires knowledge of the code base. not ideal).

What I'd suggest doing instead is creating a `PostProcessingStep` step (impl. `PipelineStep`) that maps 1-1 to output adapters / prediction types. If a user doesn't set one for the pipeline, we'll assume what we call `RAW` output. This way adapters become a great additional option to configure, rather than a potential trap.

I think this would be conceptually much cleaner, as "anything" users care about should be modular steps, not nested configs.

**Comment by agibsonccc** [comment]
@maxpumperla I'm not against this. The original intuition was focusing on making it easy to make people who are not familiar with how to do conversions of certain kinds of results (classification, coordinates etc) from ndarrays able to just configure a standard end result they understand. One of the things we want to make sure is people can use pretrained model output properly. If we can take some time to think about the non  python data scientist end user here I think we'll be fine.

**Comment by maxpumperla** [comment]
@agibsonccc agreed. this is more of a long-term thing. just wanted to put it here so we don't forget. I'll think about the structure and work on an RFC once ready
